CMakeLists.txt

```diff
@@ -514,7 +514,8 @@ macro (add_executable target)
     # invoke built-in add_executable
     # explicitly acquire and interpose malloc symbols by clickhouse_malloc
     # if GLIBC_COMPATIBILITY is ON and not sanitizer build, provide memcpy symbol explicitly to neutrialize thinlto's libcall generation.
-    if (GLIBC_COMPATIBILITY AND NOT SANITIZE)
+    # `flto=thin` used only for release build, probably this `if` should be more strict.
+    if (GLIBC_COMPATIBILITY AND NOT SANITIZE AND NOT ARCH_AMD64)
```

Comment:
why ARCH_AMD64?

Reply:
Oh thanks, incorrect flag. I was trying to fix https://clickhouse-builds.s3.yandex.net/15595/b3fa86191e4f85ad3ff4d72d6370d56b80be868c/clickhouse_special_build_check/build_log_788464832_1601815120.txt